src/zihintntl.tex

```diff
@@ -0,0 +1,202 @@
+For example, for a system with a private L1 and a shared L2, it is recommended
+that NTL.P1 and NTL.PALL indicate that temporal locality cannot be exploited by
+the L1, and that NTL.S1 and NTL.ALL indicate that temporal locality cannot be
+exploited by the L2.
+Furthermore, software tuned for such a system should use NTL.P1 to indicate
+a lack of temporal locality exploitable by the L1, or should use NTL.ALL
+indicate a lack of temporal locality exploitable by the L2.
+
+\begin{table}[h!]
+\begin{center}
+\scalebox{0.95}{
+\begin{tabular}{|l|WWWW|WWWW|}
+\hline
+Memory hierarchy & \multicolumn{4}{c|}{Recommended mapping of NTL}    & \multicolumn{4}{c|}{Recommended NTL variant for} \\
+                 & \multicolumn{4}{c|}{variant to actual cache level} & \multicolumn{4}{c|}{explicit cache management} \\
+\hline
+                                 & P1  & PALL& S1  & ALL& L1  & L2  & L3  & L4/L5  \\
+\hline
+\multicolumn{9}{|c|}{Common Scenarios} \\
+\hline
+No caches             & \multicolumn{4}{c|}{---} & \multicolumn{4}{c|}{\em none} \\
+\hline
+Private L1 only                  & L1  & L1  & L1  & L1 & ALL & --- & --- & --- \\
+Private L1; shared L2            & L1  & L1  & L2  & L2 & P1  & ALL & --- & --- \\
+Private L1; shared L2/L3         & L1  & L1  & L2  & L3 & P1  & S1  & ALL & --- \\
+Private L1/L2                    & L1  & L2  & L2  & L2 & P1  & ALL & --- & --- \\
+Private L1/L2; shared L3         & L1  & L2  & L3  & L3 & P1  & PALL& ALL & --- \\
+Private L1/L2; shared L3/L4      & L1  & L2  & L3  & L4 & P1  & PALL& S1  & ALL \\
+\hline
+\multicolumn{9}{|c|}{Uncommon Scenarios} \\
+\hline
+Private L1/L2/L3; shared L4      & L1  & L3  & L4  & L4 & P1  & P1  & PALL& ALL \\
+Private L1; shared L2/L3/L4      & L1  & L1  & L2  & L4 & P1  & S1  & ALL & ALL \\
+Private L1/L2; shared L3/L4/L5   & L1  & L2  & L3  & L5 & P1  & PALL& S1  & ALL \\
+Private L1/L2/L3; shared L4/L5   & L1  & L3  & L4  & L5 & P1  & P1  & PALL& ALL \\
+\hline
+\end{tabular}}
+\end{center}
+\caption{Mapping of NTL variants to various memory hierarchies.}
+\label{tab:ntl}
+\end{table}
+
+If the C extension is provided, compressed variants of these HINTs are also
+provided:
+C.NTL.P1 is encoded as \mbox{C.ADD {\em x0, x2}};
+C.NTL.PALL is encoded as \mbox{C.ADD {\em x0, x3}};
+C.NTL.S1 is encoded as \mbox{C.ADD {\em x0, x4}};
+and C.NTL.ALL is encoded as \mbox{C.ADD {\em x0, x5}}.
+
+The NTL instructions affect all memory-access instructions, unless otherwise
+specified.
+
+\begin{commentary}
+As of this writing, there are no exclusions to this rule, and so
+the NTL instructions affect all memory-access instructions
+defined in the base ISAs and the A, F, D, Q, C, and V standard extensions,
+as well as those defined within the hypervisor extension in Volume II.
+
+We also expect that the NTL instructions will affect some cache-maintenance
+operations.
+For example, NTL.PALL followed by a cache-line-zero instruction might indicate
```

Comment:
CBO.ZERO instead of "cache-line-zero" to be precise?

Reply:
Ack.